main.js

```diff
@@ -66,200 +104,134 @@ function displayArticles() {
     }
-    
-    const numArticlesOnPage = articles.response.docs.length;
-    if (numArticlesOnPage < 10) {
+      
+    if (currentPageArticles.length < 10) {
       nextPageButton.disabled = true;
       nextPageButton.classList.add('disabled');
     } else {
       nextPageButton.disabled = false;
       nextPageButton.classList.remove('disabled');
     }
 
-    nav.style.display = 'flex';
-    sortDiv.style.display = 'flex';
-  } else {
-    nav.style.display = 'none';
-    sortDiv.style.display = 'none';
+		searchResultsDiv.style.display = 'block';
   }
 }
 
-function displayMetaInfo() {
-  const metaInfoDiv = document.querySelector('.meta-info');
-
-  while (metaInfoDiv.firstChild) {
-    metaInfoDiv.removeChild(metaInfoDiv.firstChild);
-  }
-  
-  const totalHitsPara = document.createElement('p');
-  totalHitsPara.textContent = `Your query returned ${totalHits} hits.`;
-  metaInfoDiv.appendChild(totalHitsPara);
-}
-
 async function fetchArticles() {
   const baseURL = 'https://api.nytimes.com/svc/search/v2/articlesearch.json';
   const key = 'brtQ9fXA0I1ATPctklZe6RcanXZRklYl';
   let fullURL = `${baseURL}?api-key=${key}&page=${resultsPage}`;
   
-  const query = document.getElementById('search-term').value;
+  const query = document.getElementById('query-input').value.trim();
+  
   if (query) {
     fullURL += `&q=${query}`;
   }
 
   const beginDate = document.getElementById('begin-date').value;
+  
   if (beginDate) {
     fullURL += `&begin_date=${beginDate}`;
   }
   
   const endDate = document.getElementById('end-date').value;
+  
   if (endDate) {
     fullURL += `&end_date=${endDate}`;
   }
 
   const sortByValue = sortSelect.value;
+  
   if (sortByValue) {
     fullURL += `&sort=${sortByValue}`;
   }
-
-  let filterSubcomponents = [];
-  const newsDeskFilters = Array.from(document.getElementById('newsdesk').elements);
-  const selectedNewsDesks = newsDeskFilters.filter(newsDesk => newsDesk.checked);
-
-  if (selectedNewsDesks.length > 0) {
-    const newsDeskComponent = getNewsDeskURLComponent(selectedNewsDesks);
-    filterSubcomponents.push(newsDeskComponent);
-  }
-
-  const materialTypeFilters = Array.from(document.getElementById('material-types').elements);
-  const selectedTypes = materialTypeFilters.filter(type => type.checked);
-
-  if (selectedTypes.length > 0) {
-    const materialTypeComponent = getMaterialTypeURLComponent(selectedTypes);
-    filterSubcomponents.push(materialTypeComponent);
-  }
-
-  const location = document.getElementById('location-filter').value;
-  if (location) {
-    const locationComponent = getLocationURLComponent(location);
-    filterSubcomponents.push(locationComponent);
-  }
-
-  if (filterSubcomponents.length > 0) {
-    fullURL += getFilterURLComponent(filterSubcomponents);
+	
+  const queryFilters = getFilterValuesForURL();
+  
+  if (queryFilters.length > 0) {
+    fullURL += '&fq=';
+    
+    for (let i = 0; i < queryFilters.length; i++) {
```

Comment:
mmm, this loop looks a little old-school :) it is understandable, but I think we can do a little better. Can we work with javascript's `join` method?